project3/coordinator/coordinator_resources/messenger_manager.h

```diff
@@ -33,6 +33,13 @@ class MessengerManager {
    * @param messenger
    */
   void DeleteMessenger(const std::shared_ptr<Messenger>& messenger);
+
+  /**
+   * @brief Getter for the set of Messengers.
+   * @return The set of messengers.
+   * @note for read only.
+   */
+  const std::unordered_set<std::shared_ptr<Messenger>> GetMessengers();
```

Comment:
I assume this is supposed to return a reference?

Reply:
I'm not sure. I intended to have this return a copy of the `Messenger` set housed in the `MessengerManager`. From my understanding, the nature of `shared_ptr` would mean that returning a copy of this `unordered_set` would still allow the caller to access/modify values that the `shared_ptr` points to, since copies of the `shared_ptr` would point to the same thing. This thinking is what made me decide to add the `const`

Reply:
Hmm, yeah, I guess there is the thread-safety issue here. However, this is still a rep exposure, even with the `const`, because that only applies to the `shared_ptr` itself, not the thing it's actually pointing to. I think what you want to do instead is return a normal (non-pointer) `std::unordered_set` that is a copy of the internal set of messengers.